app/src/components/plotting.tsx

```diff
@@ -0,0 +1,223 @@
+// Plottly interface
+import { Button, Card, Form } from "react-bootstrap";
+import { useState, useRef, useEffect } from "react";
+import { socket } from "../socket";
+import { Rnd, RndResizeCallback } from "react-rnd";
+import Plot from "react-plotly.js";
+import { IoDuplicate } from "react-icons/io5";
+
+interface PlottingProps {
+  setStep: (step: number) => void;
+  setSelectedFrames: (selectedFrames: Set<number>) => void;
+}
+
+export const Plotting = ({ setStep, setSelectedFrames }: PlottingProps) => {
+  const [availablePlots, setAvailablePlots] = useState<string[]>([]);
+  const [plotData, setPlotData] = useState<{ [key: string]: any }>({});
+  const [displayedCards, setDisplayedCards] = useState<number[]>([]);
```

Comment:
pass a `useState` trigger from the sidebar to show / hide the plots window instead of the python function trigger.